src/main/java/org/commcare/xml/QueryGroupParser.java

```diff
@@ -0,0 +1,38 @@
+package org.commcare.xml;
+
+import org.commcare.suite.model.DisplayUnit;
+import org.commcare.suite.model.QueryGroup;
+import org.javarosa.xml.util.InvalidStructureException;
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.kxml2.io.KXmlParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.IOException;
+
+public class QueryGroupParser extends CommCareElementParser<QueryGroup> {
+
+    private static final String NAME_PROMPT = "group";
+    private static final String ATTR_KEY = "key";
+    private static final String NAME_DISPLAY = "display";
+
+    public QueryGroupParser(KXmlParser parser) {
+        super(parser);
+    }
+
+    @Override
+    public QueryGroup parse() throws InvalidStructureException, IOException, XmlPullParserException,
+            UnfullfilledRequirementsException {
+        checkNode("group");
+
+        String key = parser.getAttributeValue(null, ATTR_KEY);
+        DisplayUnit display = null;
+    
+        while (nextTagInBlock(NAME_PROMPT)) {
+            if (NAME_DISPLAY.equalsIgnoreCase(parser.getName())) {
+                display = parseDisplayBlock();
+            }
+        }
```

Comment:
Can you explain what this block is doing? It seems like `display` may get set multiple times here, not sure to what end.

Reply:
The suite would look like this for a `group` block. `nextTagInblock` iterates through the elements within `group` (it automatically iterates to the next element within the block without needing to increment anything in the body of the `while` loop). The first would be `display` and that whole `display` block will be parsed via `parseDisplayBlock`. The parsed values will then be used to created an instance of `QueryGroup` so that instance can be used to pull values as needed.
```
<group key="group_header_0">
    <display>
      <text>
          <locale id="search_property.m0.group_header_0"/>
      </text>
    </display>
</group>
 ```

You're right that if there were multiple `display` blocks in the `group` element, only the last `display` block will be saved. But there should be only one